### PR TITLE
Add more specific error message for invalid DateOfBirth

### LINKED ISSUE #229: Bug: add patient error message when given date that follows format but does not exist could be changed
Same issue as addAppt/delAppt. When given valid format but date does not exist for eg 2023-02-29/2024-11-31 

Error message says invalid format and should be YYYY-MM-DD.
However since it differs by a day. Doctors might have issues realising this. Perhaps a better error message such as "date does not exist" would be better if format is followed by date does not exist.

### PR DESCRIPTION
**Pull request by yong-zhuo**
fixes #229 

Changes made:
* Added a new method in `DateUtil` called `isCorrectDateFormat` which checks if the given string is in the correct date format `yyyy-MM-dd` . It does not check if the given date is
* When parsing `DateOfBirth`:
Check if given string follows the correct date format-> Check if given date is a valid date that exists -> Check if given date is in the future
* If date does not exist, the error message will display `The given date is invalid and does not exist in the calander.`

@yadobler you can consider using this method for #227

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Gender.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  */
 public class Gender {
 
-    public static final String MESSAGE_CONSTRAINTS = "Gender should be either 'M' or 'F'";
+    public static final String MESSAGE_CONSTRAINTS = "Gender should be either 'M' or 'F' and it should not be blank.";
```
Comment (PresleyChew) [inline src/main/java/seedu/address/model/person/Gender.java:12]: We might have to add this in our User Guide parameter details. 
Currently its  `Should only be either M (Male) or F (Female).` Perhaps we could add `Should not be blank` as well just like the other parameters in the parameter details. What do you think? 😄

File: src/test/java/seedu/address/commons/util/DateUtilTest.java [hunk 2]
```diff
@@ -22,10 +23,16 @@ public void isValidDate_invalidDate_returnsFalse() {
...
         assertFalse(isValidDate("2022-13-01")); // invalid month
-        assertFalse(isValidDate("2022-12-32")); // invalid day
+        assertFalse(isValidDate("2023-02-29")); // invalid day
         assertFalse(isValidDate("2022/12/12")); // invalid format
     }
 
+    @Test
+    public void isCorrectDateFormat_correctFormat_returnsTrue() {
+        assertTrue(isCorrectDateFormat("2022-12-12"));
+        assertTrue(isCorrectDateFormat("2023-02-29")); // returns true even if date does not exist
```
Comment (PresleyChew) [inline src/test/java/seedu/address/commons/util/DateUtilTest.java:33]: Good test case here!

**Review by PresleyChew — APPROVED** [review]
Overall LGTM

File: src/main/java/seedu/address/model/person/Gender.java (hunk 1 above)
Comment (yong-zhuo) [reply src/main/java/seedu/address/model/person/Gender.java:12]: I will add it to the other parameters missing 'should not be blank' in another PR 👍🏻

**Pull request merged**